server/dataSimulator.js

```diff
@@ -1,17 +1,18 @@
-var express = require('express');
 var faker = require('faker');
 var fs = require('fs');
 var csv = require('fast-csv');
-var app = express();
+var db = require('../database/database.js')
 
 // create half a million user names 
-var userGeneraton = function() {
+var userGeneration = function() {
     var users = [];
-    var usersArr = [['name', 'username', 'timezone', 'publisher']]; // to store the data here as well for writing into the csv
+    var usersArr = [];
+    //var usersArr = [['id', 'name', 'username', 'timezone', 'publisher']]; // to store the data here as well for writing into the csv
     var tempNames = {};
     var publisherProb = 0.10;
     var handle, name, number, timezone, pubBool;
-    while (users.length < 500000) {
+    var count = 1; 
+    while (count < 500000) {
         // generate name 
         handle = faker.fake("{{name.firstName}} {{name.lastName}}").toLowerCase();
```

Comment:
Try to avoid unnecessary computations, you can convert only the first letter to lowercase. Normally this wouldn't matter, but since we are generating huge datasets, any performance improvement would be good.

Reply:
good idea